service_capacity_modeling/models/org/netflix/elasticsearch.py

```diff
@@ -0,0 +1,471 @@
+    #   * Security group limits. Since you must have < 500 rules if you're
+    #       ingressing public ips)
+    #   * Maintenance. If your restart script does one node at a time you want
+    #       smaller clusters so your restarts don't take months.
+    #   * NxN network issues. Sometimes smaller clusters of bigger nodes
+    #       are better for network propagation
+    if cluster.count > (max_regional_size // zones_per_region):
+        return None
+
+    ec2_cost = zones_per_region * cluster.annual_cost
+
+    cluster.cluster_type = "elasticsearch-data"
+    clusters = Clusters(
+        total_annual_cost=round(Decimal(ec2_cost), 2),
+        zonal=[cluster] * zones_per_region,
+        regional=list(),
+    )
+
+    return CapacityPlan(
+        requirements=Requirements(zonal=[requirement] * zones_per_region),
+        candidate_clusters=clusters,
+    )
+
+
+class NflxElasticsearchCapacityModel(CapacityModel):
+    @staticmethod
+    def capacity_plan(
+        instance: Instance,
+        drive: Drive,
+        context: RegionContext,
+        desires: CapacityDesires,
+        extra_model_arguments: Dict[str, Any],
+    ) -> Optional[CapacityPlan]:
+        # (FIXME): Need elasticsearch input
+        # TODO: Use durability requirements to compute RF.
+        copies_per_region: int = _target_rf(
+            desires, extra_model_arguments.get("copies_per_region", None)
+        )
+        max_regional_size: int = extra_model_arguments.get("max_regional_size", 240)
+        max_rps_to_disk: int = extra_model_arguments.get("max_rps_to_disk", 1000)
+        # Very large nodes are hard to recover
+        max_local_disk_gib: int = extra_model_arguments.get("max_local_disk_gib", 4096)
+
+        return _estimate_elasticsearch_cluster_zonal(
+            instance=instance,
+            drive=drive,
+            desires=desires,
+            zones_per_region=context.zones_in_region,
+            copies_per_region=copies_per_region,
+            max_regional_size=max_regional_size,
+            max_local_disk_gib=max_local_disk_gib,
+            max_rps_to_disk=max_rps_to_disk,
+        )
+
+    @staticmethod
+    def description():
+        return "Netflix Streaming Elasticsearch Model"
+
+    @staticmethod
+    def extra_model_arguments() -> Sequence[Tuple[str, str, str]]:
+        return (
+            (
+                "copies_per_region",
+                "int = 3",
+                "How many copies of the data will exist e.g. RF=3. If unsupplied"
+                " this will be deduced from durability and consistency desires",
+            ),
+            (
+                "max_regional_size",
+                # Twice the size of our largest cluster
+                "int = 240",
+                "What is the maximum size of a cluster in this region",
+            ),
+            (
+                "max_local_disk_gib",
+                # Nodes larger than 4 TiB are painful to recover
+                "int = 4096",
```

Comment:
I think it makes sense to raise this slightly to include `i3en.2xlarge` which have 5,000 GB, so set `max_local_disk_gib` to `5000` to be slightly over?

Reply:
We have some clusters with i3en.3xl with 7500 GB disk space.